### PR TITLE
Fixed the signup and login issue #255

### LINKED ISSUE #255: BUG: Can't Signup
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What happened?

I tried signing up but It gives an error ( you can check in the network tab)

The API URL is wrong as it is pointing to the localhost but It should be pointed to the API deployed on production

image

### Record

- [X] I agree to follow this project's Code of Conduct
- [ ] I'm a GSSOC'24 contributor
- [X] I want to work on this issue

### PR DESCRIPTION
**Pull request by anuragnegi000**
Please include a summary of the changes and the related issue. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes #255

## Type of change

Please give a X on it which is applicable

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Refactor Code
- [ ] A documentation update
- [ ] Others(mentioned in the issue number)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

**_Test A Describe here_**

Tested this in localhost and it's working fine now

**_Test B Describe here (if Requred)_**

https://github.com/VanshKing30/FoodiesWeb/assets/115611556/176b90f0-eb83-4df1-a4a4-6bab577b8f8f

# Checklist:
give a X on it which is applicable

- [ X] My code follows the style guidelines of this project
- [ X] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [X ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by anuragnegi000** [comment]
@hustlerZzZ please merge this PR as soon as possible so everyone can Signup

**Comment by anuragnegi000** [comment]
@hustlerZzZ please add labels too

**Review by hustlerZzZ — APPROVED** [review]

**Review by aayushchugh — COMMENTED** [review]
@anuragnegi000 @hustlerZzZ Base URL is hardcoded in alot of places. We need to make sure that it's not hardcoded to `localhost` so that API calls are not made to localhost in production. It should be retrieved from `.env` file.

We can also create a custom instance of axios with our `baseurl` from `.env` file

Login API call is still made to `localhost`

image

File: src/pages/Signup.jsx [hunk 1]
```diff
@@ -77,53 +77,53 @@ function Signup() {
...
+          toast.success("Account Created Successfully!"); 
+          navigate("/home");
+        } catch (error) {
+          toast.error("Failed To Create Account. Please try again."); 
+          console.error(error); 
+        } finally {
+          setLoading(false);
+        }
+      } else {
+        const apiUrl = `http://localhost:3000/api/v1/canteenSignup`
```
[the remark this answers is not in the document]
Comment (anuragnegi000) [reply src/pages/Signup.jsx:104]: Yes this makes sense

**Pull request merged**